components/plugin_marketplace/marketplace_item/marketplace_item_app/marketplace_item_app.tsx

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+
+import type {MarketplaceLabel} from 'mattermost-redux/types/marketplace';
+
+import MarketplaceItem from '../marketplace_item';
+
+export type MarketplaceItemAppProps = {
+    id: string;
+    name: string;
+    description?: string;
+    homepage_url?: string;
+    root_url: string;
+
+    installed: boolean;
+    labels?: MarketplaceLabel[];
+
+    trackEvent: (category: string, event: string, props?: unknown) => void;
+
+    actions: {
+        installApp: (id: string, url: string) => void;
+        closeMarketplaceModal: () => void;
+    };
+};
+
+export default class MarketplaceItemApp extends React.PureComponent <MarketplaceItemAppProps> {
+    onInstall = (): void => {
+        this.props.trackEvent('plugins', 'ui_marketplace_install_app', {
+            app_id: this.props.id,
+        });
+
+        this.props.actions.installApp(this.props.id, this.props.root_url);
```

Comment:
Should we wait for any errors here, or we expect the error to appear after the marketplace closes?

Reply:
Good question. I suppose all error will show up in the UI. In my tests every failure has resulted in such.

Reply:
@larkox I dug depper into the issue and hit a wall. It seems like `executeCommand` does not correctly return an error, at least I can't `catch` it in `actions/marketplace.ts`. If you insist that we show any errors in the Marketplace modal, my proposal would be to migrate  `actions/command.js` to TS first and then further debug the error. Otherwise I would just file a ticket to show the error in the UI.

Reply:
I would say this is @aaronrothschild 's call. I feel it is a better UX to show the error in the marketplace, but I am not sure.

Reply:
I don't know if there is a UX rule that says that an error should always be shown in the user action context, but I believe that normally the error should be shown in the modal.

Reply:
@larkox @catalintomai I've updated the PR in 201079754c35b84881edae0ec7e4ca184ea46ef7 to show any errors thrown by the slash command to be shown. Please take a look a the updated screenshots.